### ISSUE TITLE
Factorio crash with event system (user error)

### ISSUE THREAD

**Issue by Nexela** [Afforess/Factorio-Stdlib#51]
Hard crash with Factorio using event system. Not super well versed in Lua/stdlib so I am posting this here for your input in case it is something preventable code wise.

I made a boo-boo when changing some stuff over to use STDLIB event system

Event.register(Event.core_events.init, myfunction()) --Hard crash when it executes, note the () on the end of it. 

Calling it correctly does work as intended :)
Event.register(Event.core_events.init, myfunction)

**Comment by Afforess** [comment]
So essentially, the issue is that you are calling your function before the registration occurs, except without the event object, so your the tries to run and the game, the event, and nothing else associated with the callback exists and you crash. Essentially this is the same problem as accessing `game` before `on_init` occurs.

There isn't a lot that can be done here, because:
* You called the function at loadup, stdlib never even had a chance register the event
* This occurs before any events or initialization

In short, there is nothing stdlib can do about typos or other errors that occur at such an early stage. On the plus side, the crash occurs pretty much immediately here, so you should get the line number and be able to debug easily.

**Comment by Nexela** [comment]
> You called the function

Well that was a big brain fart on my end.  I didn't even think about myfunction() being executed first. 

Thanks for the explanation and the awesome library of functions.
